riscv/src/continuations/bootloader.rs

```diff
@@ -22,6 +22,26 @@ pub const MEMORY_HASH_START_INDEX: usize = 2 * REGISTER_NAMES.len();
 pub const NUM_PAGES_INDEX: usize = MEMORY_HASH_START_INDEX + 8;
 pub const PAGE_INPUTS_OFFSET: usize = NUM_PAGES_INDEX + 1;
 
+/// Computes an upper bound of how long the shutdown routine will run, for a given number of pages.
+pub fn num_shutdown_routine_upper_bound(num_pages: usize) -> usize {
```

Comment:
```suggestion
pub fn shutdown_routine_upper_bound(num_pages: usize) -> usize {
```

I guess `upper_bound` is already precise enough